Mage/src/main/java/mage/util/functions/CopyTokenFunction.java

```diff
@@ -90,8 +90,8 @@ public Token apply(Card source, Game game) {
             target.addAbility(ability);
         }
 
-        target.getPower().modifyBaseValue(sourceObj.getPower().getBaseValueModified());
-        target.getToughness().modifyBaseValue(sourceObj.getToughness().getBaseValueModified());
+        target.getPower().setModifiedBaseValue(sourceObj.getPower().getBaseValue());
+        target.getToughness().setModifiedBaseValue(sourceObj.getToughness().getBaseValue());
```

Comment:
This is also making a copy of an actual permanent, so I also think this should be getBaseValue and not getModifiedBaseValue.

Reply:
So the current failures are because of `.getBaseValue` instead of `.getModifiedBaseValue`. The failing tests are copying a token that's a copy of another creature. Those tokens have `.getBaseValues` of 0 for their toughness, but a non-zero value for the modifiedBaseValue. I need to see if that's supposed to happen.

Reply:
it shouldn't, the token's copiable values should be the same as whatever it's copying unless the effect creating it says otherwise (like [[Croaking Counterpart]])

Reply:
Alright. So the issue goes like this following. When making a token, this function will set the modified value of the power and toughness, but leaves the base values to 0. So when that token then gets copied, it will set the copy's modified value to 0.

This will probably also create issues when anything else tries to modify the base value since there won't be anything to fall back to when the effect ends.

This should be setting the unmodified base value (I'll need to make a `setBaseValue()` function in MageInt.
The same problem affects several cards

Reply:
For example, the Scarab God:
https://github.com/magefree/mage/blob/f827a344329bd27ad8f1b3d4b96f2b598089178d/Mage.Sets/src/mage/cards/g/GodPharaohsGift.java#L86-L90


this should be setting the base P/T to 4/4, and not the modified base values. When these tokens are copied, they should be copied as 4/4. When something like Biomass Mutation is cast their base power toughness needs to go back to a 4/4 after the end of the turn.

Reply:
This `CopyTokenFunction` class is only used by `CardUtil.copyTo`, and every single card and ability that uses that method first creates an EmptyToken and then calls `copyTo` to add abilities to it. In other words, the `token` argument of  We should probably refactor so that CardUtil directly creates a token with all the desired characteristics and abilities, instead of needing the user to supply an empty token.

Reply:
> This `CopyTokenFunction` class is only used by `CardUtil.copyTo`, and every single card and ability that uses that method first creates an EmptyToken and then calls `copyTo` to add abilities to it. In other words, the `token` ctor argument / field of `CopyTokenFunction` is always an empty token! We should probably refactor this mess so that CardUtil directly creates a token with all the desired characteristics and abilities, instead of needing the user to supply an empty token.

Do you want me to see about doing that as part of this PR, or do that in a subsequent one?

Reply:
> Do you want me to see about doing that as part of this PR, or do that in a subsequent one?

It's very low priority.

Reply:
> > Do you want me to see about doing that as part of this PR, or do that in a subsequent one?
> 
> It's very low priority, cosmetic really.

Though I was really scratching my head at a line like `CardUtil.copyTo(token).from(cardChosen, game)`.

Okay, I'll open a separate issue for it.